grove/amplification/grover.py

```diff
@@ -61,41 +66,51 @@ def _construct_grover_circuit(self):
         oracle = pq.Program()
         oracle_name = "GROVER_ORACLE"
         oracle.defgate(oracle_name, self.unitary_function_mapping)
-        qubits = [oracle.alloc() for _ in range(int(self.n_qubits / 2))]
-        oracle.inst(tuple([oracle_name] + qubits))
-        self.grover_circuit = self.oracle_grover(oracle, qubits)
+        oracle.inst(tuple([oracle_name] + self.qubits))
+        self.grover_circuit = self.oracle_grover(oracle, self.qubits)
 
     def _init_attr(self, bitstring_map):
         """Initializes an instance of Grover's Algorithm given a bitstring_map.
 
-        :param dict bitstring_map: dict with string keys corresponding to bitstrings, and integer
+        :param bitstring_map: dict with string keys corresponding to bitstrings, and integer
          values corresponding to the desired phase on the output state.
+        :type bitstring_map: dict[str, int]
```

Comment:
why do we suddenly change the formatting of how we annotate types?

Reply:
@jotterbach So I had a lot (not all) of them written in the python 3 typing format how you had them, but that doesn't link with the type in the documentation. It doesn't make sense to me that when you document the type in the docstring you give the type in the `typing` module.

Reply:
I agree, but let's not just randomly start to change this in select places. If we change it we should at least do this module wide, better even across the full code-base. I'd like to keep consistency.

Reply:
@jotterbach That's fine. Is it consistent right now? Part of the reason I didn't feel strongly against changing is I thought it wasn't consistent.

Reply:
In any case, moving forward I think this is the right way to do it- would you rather I change them back and do a module/package-wide update later?

Reply:
I'd prefer that, yes, because that makes the changes more atomic and contained into dedicated units.

Reply:
aye aye

Reply:
done